src/components/portrait/portrait.test.js

```diff
@@ -0,0 +1,289 @@
+import React from "react";
+import Portrait from "./portrait.component";
+import Box from "../box";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { icon, getDataElementByValue } from "../../../cypress/locators";
+
+import {
+  portraitPreview,
+  portraitInitials,
+  portraitAlt,
+} from "../../../cypress/locators/portrait";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testImage = [
+  "https://www.gravatar.com/avatar/cceff1ad774bf89b1f75cb6e5005333c?s=40&d=404",
+];
+const imageUrl = [
+  "https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight&accessoriesType=Blank&hairColor=BrownDark&facialHairType=Blank&clotheType=BlazerShirt&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light",
+  "https://www.gravatar.com/avatar/05c1c705ee45d7ae88b80b3a8866ddaa?s=24&d=404",
+];
+
+const iconlist = [
+  "filter",
+  "error",
+  "warning",
+  "info",
+  "key",
+  "add",
+  "admin",
+  "alert",
+];
+
+const PortraitComponent = ({ ...props }) => {
+  return <Portrait {...props} />;
+};
+
+const TooltipPortraitComponent = ({ ...props }) => {
+  return (
+    <Portrait
+      tooltipMessage="Rebecca Smith"
+      tooltipIsVisible
+      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight&accessoriesType=Blank&hairColor=BrownDark&facialHairType=Blank&clotheType=BlazerShirt&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
+      {...props}
+    />
+  );
+};
+
+context("Tests for Portrait component", () => {
+  describe("check props for Portrait component", () => {
+    it.each(testImage)(
+      "should check gravatar for Portrait components",
+      (srcImage) => {
+        CypressMountWithProviders(
+          <PortraitComponent gravatar="chris.barber@sage.com" />
+        );
+        portraitPreview()
+          .should("have.css", "height", "40px")
+          .and("have.css", "width", "40px");
+        portraitPreview().children().should("have.attr", "src", srcImage);
+      }
+    );
+
+    it.each(imageUrl)(
+      "should check custom image url for Portrait component",
+      (srcImage) => {
+        CypressMountWithProviders(<PortraitComponent src={srcImage} />);
+        portraitPreview()
+          .should("have.css", "height", "40px")
+          .and("have.css", "width", "40px");
+        portraitPreview().children().should("have.attr", "src", srcImage);
+      }
+    );
+
+    it.each([
+      ["XS", 24],
```

Comment:
we can import those values from `src/components/portrait/portrait.config.ts` instead of declaring them here